python/ray/tune/hpo_scheduler.py

```diff
@@ -0,0 +1,201 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import os
+import copy
+import numpy as np
+try:
+    import hyperopt as hpo
+except Exception as e:
+    print("HyperOpt not found; perhaps not installed?")
+    raise e
+
+from ray.tune.trial import Trial
+from ray.tune.error import TuneError
+from ray.tune.registry import register_trainable
+from ray.tune.trial_scheduler import TrialScheduler, FIFOScheduler
+from ray.tune.config_parser import make_parser
+from ray.tune.variant_generator import to_argv
+
+
+class HyperOptScheduler(FIFOScheduler):
+
+    def __init__(self, max_concurrent=10, loss_attr="mean_loss"):
+        self._max_concurrent = max_concurrent  # NOTE: this is modified later
```

Comment:
Shouldn't we always launch as many trials at once as resources permits?

Reply:
Not necessarily; the more concurrent trials the less effective your search. For example, SigOpt only allows for 10 concurrent trials.

Now, for this particular case it could make sense to default to as many as resource permits.